keps/sig-windows/2578-windows-conformance/README.md

```diff
@@ -0,0 +1,419 @@
+- Ability to protect IPv4 pods from acceessing other pods when TCP NetworkPolicys are present that block specific namespace connectibity.
+
+#### Windows Service Proxying
+
+- Ability to support IPv6 and IPv4 services for accessing internal Linux pods from Windows pods over ClusterIP Endpoints.
+- Ability to support IPv6 and IPv4 services for accessing internal Linux pods from Windows pods over NodePort Endpoints.
+- Ability to support IPv6 and IPv4 services for accessing internal Windows pods from Linux pods over ClusterIP Endpoints.
+- Ability to support IPv6 and IPv4 services for accessing internal Windows pods from Linux pods over NodePort Endpoints.
+
+### Implementation
+
+This section details the implementation of a "tool" which verifies these clusters must allow for the following, differentiated set of verifications, specified above.
+
+- core conformance
+- hostProcess
+- activeDirectory
+- networkPolicy
+- serviceProxy
+
+#### Option 1 - Test tags
+
+One way to implement such a tool is to, specifically, target all available "ginkgo" tests in Kubernetes core itself, and run these one at a time (or concurrently).  Not we *do not* assert that these tests *must* run in parallel in order to comprise a conformance clusters (although this is often done in CI, its generally not a requirement of Kubernetes Conformance for linux clusters, either, since cluster capacity is not related to the functionality of a cluster). As an example, one such implementation of such a Conformance tool might be:
+
+```
+--ginkgo.focus=\\[Conformance\\]|\\[NodeConformance\\]|\\[sig-windows\\]|\\[sig-apps\\].CronJob --ginkgo.skip=\\[LinuxOnly\\]|\\[k8s.io\\].Pods.*should.cap.back-off.at.MaxContainerBackOff.\\[Slow\\]\\[NodeConformance\\]|\\[k8s.io\\].Pods.*should.have.their.auto-restart.back-off.timer.reset.on.image.update.\\[Slow\\]\\[NodeConformance\\]"
+```
+
+Currently, the de facto standard for windows testing resides at https://github.com/kubernetes-sigs/windows-testing, and we suggest this tag to users, however it should be noted that this does *not* conform to the specification in this document:
+
+- it triggers some tests which uses non `pause` images for validating functionality.
+- it doesn't implement the pod churn or emptyDir sharing tests.
+- it doesn't implement the *reboot* tests which are defined in this specification.
+
+The ability to specify HostProcess, NetworkPolicy, and other tests in this document would follow a similar tagging strategy, and any such tool could just invoke the existing Kubernetes end-to-end test binary with appropriate `focus` and `skip` arguments.
+
+#### Option 2 - Sonobuoy implementation for convenience
+
+Given the complexity of the above tagging scheme, we propose a concrete example of a binary program for Windows verification.
+
+```
+sonobuoy run --mode=windows-conformance --windows.hostProcess --windows.activeDirectory ...
```

Comment:
im not at all committed to this implementation, its a good example though of how we can define windows
operational readiness w/o forcing OS versions, bc the non-standard/non-legacy features can be added on.

Reply:
I like the simplicity of this but we will need somewhere to define what those buckets contain so they can be used across CI and conformance and be tied to a particular k8s release.

Reply:
i think this is solved since, sonobuoy gives you the full e2e test results/report in junit format right?